pkg/controller/nodecsrapprover/node_csr_approver_test.go

```diff
@@ -17,102 +17,706 @@ limitations under the License.
-IDjKG4ahwgDJQRtpmGqufjFBqrVVI3DxEFvt2RATJ3HA
------END CERTIFICATE REQUEST-----`
+dGVybmFsMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEI0vty8uehZOaXxjluaEi
+DaTBhB0rplD92+Mi8Xt36nCq+BmrtXBIst1Cp8jzx/AESEuC/06T1FoZnXir5H5k
+KKBPME0GCSqGSIb3DQEJDjFAMD4wPAYDVR0RBDUwM4IraXAtMTcyLTMxLTExNC00
+OC5ldS1lYXN0LTMuY29tcHV0ZS5pbnRlcm5hbIcEwAACGDAKBggqhkjOPQQDAgNH
+ADBEAiARKzqSf5A4K9mSpfN5BZkvagN2hpSuCrNh969aMHKWGgIgLMlqn6GED5cS
+rHy6Ox5XuLIEfuyO2lw4Tpse1T0MN5c=
+-----END CERTIFICATE REQUEST-----
+`
+
+	testInvalidIPAddressCSR = `-----BEGIN CERTIFICATE REQUEST-----
+MIIBYzCCAQoCAQAwWTEVMBMGA1UEChMMc3lzdGVtOm5vZGVzMUAwPgYDVQQDEzdz
+eXN0ZW06bm9kZTppcC0xNzItMzEtMTE0LTQ4LmV1LXdlc3QtMy5jb21wdXRlLmlu
+dGVybmFsMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE5bGjubaUuMpo5IHK/nfm
+fPoG/uPDkDUtx9wugPaNGd8vn5Mu/3QMf3NuD5kXRY4RCWl9w4vddkLdT+hSekP1
+i6BPME0GCSqGSIb3DQEJDjFAMD4wPAYDVR0RBDUwM4IraXAtMTcyLTMxLTExNC00
+OC5ldS13ZXN0LTMuY29tcHV0ZS5pbnRlcm5hbIcEwAACAjAKBggqhkjOPQQDAgNH
+ADBEAiB9myW6lAcQKX2WCsfriwK3aIMzgY4jCkr8tPV1oZeLLQIgbwlqgQcAOQqC
+/6Fr0hK/AIDlykBtDKkNYDgm3n3Oqqc=
+-----END CERTIFICATE REQUEST-----
+`
+
+	testAdditionalDNSNameCSR = `-----BEGIN CERTIFICATE REQUEST-----
+MIIBkTCCATcCAQAwWTEVMBMGA1UEChMMc3lzdGVtOm5vZGVzMUAwPgYDVQQDEzdz
+eXN0ZW06bm9kZTppcC0xNzItMzEtMTE0LTQ4LmV1LXdlc3QtMy5jb21wdXRlLmlu
+dGVybmFsMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEA9/qlNDkHRRv/MMvqGDv
+glwVIepZ1nXpA72XmwT9eKGKIGhvYkU5y6HkpkLv45niTxhhYN4EbSwAC7651ZsK
+6qB8MHoGCSqGSIb3DQEJDjFtMGswaQYDVR0RBGIwYIIraXAtMTcyLTMxLTExNC00
+OC5ldS13ZXN0LTMuY29tcHV0ZS5pbnRlcm5hbIIraXAtMTcyLTMxLTExNC00OC5l
+dS1lYXN0LTMuY29tcHV0ZS5pbnRlcm5hbIcEwAACGDAKBggqhkjOPQQDAgNIADBF
+AiAr5gSITnVYXOFmvEU+mcHk8w/8/cbKwMHWtLZ0ZLN+igIhAP0BE//qz0lAFINS
+Su19WQIpe9VR/dGJvXfsgaLjyPyJ
+-----END CERTIFICATE REQUEST-----
+`
+
+	testAdditionalIPAddressCSR = `-----BEGIN CERTIFICATE REQUEST-----
+MIIBazCCARACAQAwWTEVMBMGA1UEChMMc3lzdGVtOm5vZGVzMUAwPgYDVQQDEzdz
+eXN0ZW06bm9kZTppcC0xNzItMzEtMTE0LTQ4LmV1LXdlc3QtMy5jb21wdXRlLmlu
+dGVybmFsMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEPg/O69m6AjfXB8XoKjAx
+2RqAdKzt1aJ9u0rbOcoa2QkTcFeMY2GbAZ/6IVWnTDsqORJeJrHP5WYNUFgBjQ2c
+gaBVMFMGCSqGSIb3DQEJDjFGMEQwQgYDVR0RBDswOYIraXAtMTcyLTMxLTExNC00
+OC5ldS13ZXN0LTMuY29tcHV0ZS5pbnRlcm5hbIcEwAACGIcEwAACGTAKBggqhkjO
+PQQDAgNJADBGAiEA3HDpVRYYgcmdCzq5o6mwkMzegZ0P0aZNPdCQLyJt3GoCIQDo
+/iL1+piFJXAOI2GjsZNpeQJ4rPJ7l/t95tzgcVAUtw==
+-----END CERTIFICATE REQUEST-----
+`
+
+	testNoDNSNameCSR = `-----BEGIN CERTIFICATE REQUEST-----
+MIIBNjCB3QIBADBZMRUwEwYDVQQKEwxzeXN0ZW06bm9kZXMxQDA+BgNVBAMTN3N5
+c3RlbTpub2RlOmlwLTE3Mi0zMS0xMTQtNDguZXUtd2VzdC0zLmNvbXB1dGUuaW50
+ZXJuYWwwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAATjbIZVQA/A+lki63bMCgVR
+UrkCqtVufWtKU5KPTD9BGZ6/dNW/uUdVftLe7Vp4+8cp2YoZIQA5NfMGYMtcE9PV
+oCIwIAYJKoZIhvcNAQkOMRMwETAPBgNVHREECDAGhwTAAAIYMAoGCCqGSM49BAMC
+A0gAMEUCIQDxZpheeDW9azYm0T0OoOyFCFslJsRf9uvJM66byWr0KwIgduhJVRog
+ELloyxg3KHxPJi8TYkTPkB+Pc9XiUGmhkzE=
+-----END CERTIFICATE REQUEST-----
+`
 )
 
-func TestReconciler_Reconcile(t *testing.T) {
```

Comment:
@irozzo-1A @kron4eg I have dropped this test because:
* I was struggling with getting the fake client working
* I'm not sure how to handle reactions correctly
* This test might not be needed since we already test all validation cases